keras/models.py

```diff
@@ -1357,8 +1406,10 @@ def fit_generator(self, generator, samples_per_epoch, nb_epoch,
             callbacks: list of callbacks to be called during training.
             validation_data: dictionary mapping input names and outputs names
                 to appropriate numpy arrays to be used as
-                held-out validation data.
-                All arrays should contain the same number of samples.
+                held-out validation data, or a generator yielding such
+                dictionaries. All arrays should contain the same number
+                of samples. If a generator, a single yield will be used
```

Comment:
Having `evaluate_generator` seems reasonable. It would not be easily feasible to include generator support in `fit` though (since the training data is split into two arguments).